src/test/java/com/linecorp/thrift/plugin/ThriftPluginTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.thrift.plugin;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Set;
+
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+import org.gradle.testkit.runner.TaskOutcome;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class ThriftPluginTest {
+    @TempDir
+    Path projectDir;
+
+    private Path buildFile;
+
+    private Path thriftPath;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        buildFile = projectDir.resolve("build.gradle");
+        Files.writeString(buildFile,
+                          "plugins { \n"
+                          + "id \"java\" \n"
+                          + "id \"org.jruyi.thrift\" \n"
+                          + "}\n",
+                          StandardOpenOption.CREATE);
+
+        Files.createDirectories(projectDir.resolve("src/main/thrift"));
+        Files.copy(Path.of("src/test/resources/test.thrift"),
+                   projectDir.resolve("src/main/thrift/test.thrift"));
+
+        thriftPath = Paths.get("lib/thrift-0.17.0");
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "7.6" })
```

Comment:
We are using `@ParameterizedTest` to support running tests for multiple versions of Gradle in the future?

Reply:
Yes